### PR TITLE
early termination of CUDA tests 

### PR DESCRIPTION
**Pull request by walterddr**
This is follow up on #49799.

* uses `torch.cuda.synchronize()` to validate CUDA assert instead of inspecting error message.
* remove non CUDA tests.
* fixed slow_tests fails on master but not normal test issue

### CODE REVIEW

File: test/test_testing.py [hunk 1]
```diff
@@ -438,6 +438,54 @@ def test_assert_messages(self, device):
...
+from torch.testing._internal.common_utils import (TestCase, run_tests)
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+
+# This test is added to ensure that test suite terminates early when
+# CUDA assert was thrown since all subsequent test will fail.
+# See: https://github.com/pytorch/pytorch/issues/49019
+# This test file should be invoked from test_testing.py
+class TestThatContainsCUDAAssertFailure(TestCase):
+
+    def test_throw_unrecoverable_cuda_exception(self, device):
```
Comment (walterddr) [inline test/test_testing.py:462]: this also needs to be decorated with `@slowTest` because otherwise the test will be ignored.

**Review by walterddr — COMMENTED** [review]
lol finally figured out why it only fails on slow test but runs fine on all others. `@slowtest` decorator also prevents test from running in the inner test cases.

File: test/test_testing.py [hunk 2]
```diff
@@ -438,6 +438,56 @@ def test_assert_messages(self, device):
...
+
+if __name__ == '__main__':
+    run_tests()
+"""
+
+        # Test running of cuda assert test suite should early terminate.
+        p = subprocess.run([sys.executable, '-c', problematic_test_script], stderr=subprocess.PIPE, timeout=120)
+        # should capture CUDA error
+        self.assertIn('CUDA error: device-side assert triggered', p.stderr.decode('ascii'))
+        # should run only 3 tests - 2 CPUs and 1 CUDA (remaining CUDA test should skip)
```
Comment (mruberry) [inline test/test_testing.py:488]: This comment is outdated ("should run only 1 test ...")

**Review by mruberry — APPROVED** [review]
Nice job, @walterddr! I made one inline comment about updating an old comment.

File: test/test_testing.py (hunk 2 above)
Comment (walterddr) [inline test/test_testing.py:488]:
```suggestion
        # should run only 1 test because it throws unrecoverable error.
```

**Pull request closed without merging**